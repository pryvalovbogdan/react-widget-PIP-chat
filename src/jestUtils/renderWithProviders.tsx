import React, { PropsWithChildren } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { I18nextProvider } from 'react-i18next';

import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';

import i18next from './i18nextJest.ts';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  route?: string;
}

function renderWithProviders(
  component: React.ReactElement,
  { route = '', ...renderOptions }: ExtendedRenderOptions = {},
) {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return (
      <Router>
        <I18nextProvider i18n={i18next}>{children}</I18nextProvider>
      </Router>
    );
  }

  return { ...render(component, { wrapper: Wrapper, ...renderOptions }) };
}

export { renderWithProviders };
