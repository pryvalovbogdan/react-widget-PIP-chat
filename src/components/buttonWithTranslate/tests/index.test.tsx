import { renderWithProviders } from '@jestUtils/renderWithProviders.tsx';

import { ButtonWithTranslate } from '../index.tsx';

describe('Test ButtonWithTranslate', () => {
  test('snapshot', () => {
    const { baseElement } = renderWithProviders(
      <ButtonWithTranslate i18Value={2} handle={() => {}} i18Key='count' backgroundColor='red' />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
