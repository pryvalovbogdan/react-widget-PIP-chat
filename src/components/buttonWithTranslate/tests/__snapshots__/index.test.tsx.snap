// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonWithTranslate snapshot 1`] = `
<body>
  <div>
    <button
      style="background-color: red;"
    >
      count is 2
    </button>
  </div>
</body>
`;
